txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/BridgeXAResource.java

```diff
@@ -70,7 +71,7 @@
      * @param externalTxId the parent JTA transaction identifier.
      * @param bridgeWrapper the control for the subordinate WS-AT transaction.
      */
-    public BridgeXAResource(Uid externalTxId, BridgeWrapper bridgeWrapper)
+    public BridgeXAResource(Transaction externalTxId, BridgeWrapper bridgeWrapper)
```

Comment:
Not changing the old constructor will mean any client code (that is not under our control) will still compile. However, unless you can somehow map from a Uid to a transaction then client code will fail at run time.